admin/cron.php

```diff
@@ -17,66 +17,74 @@
+    fclose($logdescriptor);
 });
 
 // zapnout zobrazení chyb
-ini_set('display_errors',   true); // zobrazovat chyby obecně
-ini_set('error_reporting',  E_ALL ^ E_STRICT); // vybrat typy chyb k zobrazení
-ini_set('html_errors',      false); // chyby zobrazovat jako plaintext
+ini_set('display_errors', true); // zobrazovat chyby obecně
+ini_set('error_reporting', E_ALL ^ E_STRICT); // vybrat typy chyb k zobrazení
+ini_set('html_errors', false); // chyby zobrazovat jako plaintext
 
 /////////////////////////////////// cron kód ///////////////////////////////////
 
 logs('Začínám provádět cron script.');
 
 if (defined('FIO_TOKEN') && FIO_TOKEN !== '') {
-  logs('Zpracovávám nové platby přes Fio API.');
-  $platby = Platby::nactiNove();
-  foreach($platby as $p) {
-    logs('platba ' . $p->id() . ' (' . $p->castka() . 'Kč, VS: ' . $p->vs() . ($p->zprava() ? ', zpráva: ' . $p->zprava() : '') . ')');
-  }
-  if(!$platby) logs('Žádné zaúčtovatelné platby.');
-}
-else {
-  logs('FIO_TOKEN není definován, přeskakuji nové platby.');
+    logs('Zpracovávám nové platby přes Fio API.');
+    $platby = Platby::nactiNove();
+    foreach ($platby as $p) {
+        logs('platba ' . $p->id() . ' (' . $p->castka() . 'Kč, VS: ' . $p->vs() . ($p->zprava() ? ', zpráva: ' . $p->zprava() : '') . ')');
+    }
+    if (!$platby) logs('Žádné zaúčtovatelné platby.');
+} else {
+    logs('FIO_TOKEN není definován, přeskakuji nové platby.');
 }
 
 logs('Odemykám zamčené aktivity.');
 $i = Aktivita::odemciHromadne();
 logs("Odemčeno $i aktivit.");
 
-
-if(date('G') == 5) { // 5 hodin ráno
-  logs('Zálohuji databázi.');
-
-  if(!defined('ZALOHA_DB_SLOZKA') || !ZALOHA_DB_SLOZKA) {
-      logs('Není definována konstanta s adresářem pro zálohování.');
-  } else {
-      $db = dbConnect();
-      $dump = new MySQLDump($db);
-      $time = date('Y-m-d_His');
-      $dump->save(ZALOHA_DB_SLOZKA . "/export_$time.sql.gz");
-      logs('Záloha databáze dokončena.');
-  }
+if (date('G') == 5) { // 5 hodin ráno
+    logs('Zálohuji databázi...');
+    $chybaZalohovaniDb = null;
+    if (!defined('ZALOHA_DB_SLOZKA') || !ZALOHA_DB_SLOZKA) {
+        $chybaZalohovaniDb = 'Není definována konstanta s adresářem pro zálohování ZALOHA_DB_SLOZKA.';
+    } elseif (!is_dir(ZALOHA_DB_SLOZKA)
+        && !@mkdir(ZALOHA_DB_SLOZKA, 0750, true)
+        && !is_dir(ZALOHA_DB_SLOZKA)
+    ) {
+        $chybaZalohovaniDb = "Nelze vytvořit adresář pro zálohování '" . ZALOHA_DB_SLOZKA . "': " . implode("\n", (array)error_get_last());
+    }
+    if ($chybaZalohovaniDb) {
+        logs($chybaZalohovaniDb);
+        (new GcMail)
+            ->adresat('info@gamecon.cz')
+            ->predmet('Neproběhla záloha databáze ' . date(\Gamecon\Cas\DateTimeCz::FORMAT_DB))
+            ->text($chybaZalohovaniDb)
+            ->odeslat();
+    }
+    $dump = new MySQLDump(dbConnect());
+    $time = date('Y-m-d_His');
+    $dump->save(ZALOHA_DB_SLOZKA . "/export_$time.sql.gz");
```

Comment:
Nevím, jestli to odeslání e-mailu nechceš pozdržet až za tento bod. `save` mimochodem při neúspěchu vyhazuje výjimku, kterou (pravděpodobně?) chceš odchytit a zalogovat.

Reply:
Máš recht. Hodil jsem to až za to. Když paranoia, tak absolutní.